src/Components/Server/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs

```diff
@@ -71,15 +68,8 @@ public ValueTask SetAsync(string key, object value)
     /// <returns>A <see cref="ValueTask"/> representing the completion of the operation.</returns>
     public ValueTask SetAsync(string purpose, string key, object value)
     {
-        if (string.IsNullOrEmpty(purpose))
-        {
-            throw new ArgumentException("Cannot be null or empty", nameof(purpose));
-        }
-
-        if (string.IsNullOrEmpty(key))
-        {
-            throw new ArgumentException("Cannot be null or empty", nameof(key));
-        }
+        ArgumentException.ThrowIfNullOrEmpty(purpose);
```

Comment:
Observable change - not saying good or bad, but: definitely a change; I'm fine with it personally

Reply:
I'm missing what observable here?

Reply:
The specific error message will change. Very much a pedantic "we should at least be aware that we're changing something, even if it is something that nobody should reasonably rely on", and in this case very much in the "trivial, no custom messages, just go for it" category.

Reply:
Also note param name that gets blamed: "key" vs "purpose"

Reply:
Effectively, we do not have an overload that accepts a custom message.
I just open a [Api Proposal on dotnet/runtime](https://github.com/dotnet/runtime/issues/86833) to have the option to set the custom message.
we can just update only those part with generic message

Reply:
Ah, right. By precedent, I think we've had no problem changing exception messages, including in cases like this, so long as that doesn't lose actual useful information.